src/test/java/com/imgix/test/TestAll.java

```diff
@@ -28,6 +28,16 @@ public void testURLBuilderRaisesExceptionOnNoDomains() {
 		URLBuilder ub = new URLBuilder(new String ());
 	}
 
+	@Test
+	public void testURLBuilderUsesHttpsByDefault() {
+		URLBuilder ub = new URLBuilder("example.imgix.net");
+		assertEquals("https://example.imgix.net/?ixlib=java-2.1.1", ub.createURL(""));
+		// Set `useHttps` to false.
+		ub.setUseHttps(false);
+		assertNotEquals("https://example.imgix.net/?ixlib=java-2.1.1", ub.createURL(""));
+		assertEquals("http://example.imgix.net/?ixlib=java-2.1.1", ub.createURL(""));
+	}
+
```

Comment:
1) Let's not pass just an empty string to `createURL`, the tests should ideally mimic actual user behavior.
2) It looks like this one test is testing two separate things. But it also made me realize that we don't have a test for `setUseHttps`, which we should. So let's break it apart into 3 cases:

-  `testURLBuilderUsesHttpsByDefault()`: just L33-34
-  `testURLBuilderOverridesHttpsByDefault()`: using `new URLBuilder("example.imgix.net", false);`
-  `testsetUseHttps()`: L36

Reply:
1. Okay, I saw passing an empty string as the "base case." Passing an empty string is allowable, and _could_ mimic user behavior, no?
2. I agree. This test tests way more than `useHttps`. This  test is testing more than two separate things. `createUrl(_)` calls `createUrl(_, _, _)`,  which in turn calls `URLHelper`. It is essentially an integration test at this point.
3. This is kind of what I meant by, "this behavior isn't easy to test [in isolation] given the current API" (paraphrasing).
4. I like the idea of splitting this up, what do you think about this?

```java
@Test
public void testURLBuilderUsesHttpsByDefault() {
}

@Test
public void testSetUseHttpsFalse() {
}

@Test
public void testSetUseHttpsTrue() {
}
```

Reply:
1. Yes it's allowable, but it would result in a URL with no path and therefore wouldn't actually resolve to an image.
2. These tests look good, please include an additional one for passing in `false` to the constructor.

Reply:
1. Okay, I will add a path. However, we are already testing "too much" and I thought that passing an empty string was closest to testing the most basic url that can be default-constructed (since the interface does not allow for that kind of thing explicitly). I will add a path. Would you like me to test a url and path that resolve to an actual image?
2. Got it, but why stop at the two-parameter constructor? There's still the 3 and 4-parameter constructors to handle which brings me to 
3. Before we grow our test suite we should discuss a few more things.